library/x509_crt.c

```diff
@@ -1449,8 +1449,8 @@ int mbedtls_x509_crt_parse_path(mbedtls_x509_crt *chain, const char *path)
         }
 
         w_ret = WideCharToMultiByte(CP_ACP, 0, file_data.cFileName,
-                                    lstrlenW(file_data.cFileName),
-                                    p, (int) len - 1,
+                                    -1,
+                                    p, (int) len,
```

Comment:
With this change, `WideCharToMultiByte()` finds and processes the terminating null itself, and outputs it, so we can give it the full buffer (i.e. `len` rather than `len - 1`)

See https://learn.microsoft.com/en-us/windows/win32/api/stringapiset/nf-stringapiset-widechartomultibyte